docs/general/builtins/table-functions.rst

```diff
@@ -198,6 +198,149 @@ arrays within the same level.
+    SELECT 1 row in set (... sec)
+
+::
+
+    cr> select regexp_matches('alcatraz', 'traz') as matched;
+    +----------+
+    | matched  |
+    +----------+
+    | ["traz"] |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+Through array element access functionality, a group can be selected directly.
+See :ref:`sql_dql_object_arrays_select` for details.
+
+::
+
+    cr> select regexp_matches('alcatraz', '(a(.+)z)')[2] as second_group;
+    +--------------+
+    | second_group |
+    +--------------+
+    | lcatra       |
+    +--------------+
+    SELECT 1 row in set (... sec)
+
+.. _table-functions-regexp-matches-flags:
+
+Flags
+.....
+
+This function takes a number of flags as optional third parameter. These flags
+are given as a string containing any of the characters listed below. Order does
+not matter.
+
++-------+---------------------------------------------------------------------+
+| Flag  | Description                                                         |
++=======+=====================================================================+
+| ``i`` | enable case insensitive matching                                    |
++-------+---------------------------------------------------------------------+
+| ``u`` | enable unicode case folding when used together with ``i``           |
++-------+---------------------------------------------------------------------+
+| ``U`` | enable unicode support for character classes like ``\W``            |
++-------+---------------------------------------------------------------------+
+| ``s`` | make ``.`` match line terminators, too                              |
++-------+---------------------------------------------------------------------+
+| ``m`` | make ``^`` and ``$`` match on the beginning or end of a line        |
+|       | too.                                                                |
++-------+---------------------------------------------------------------------+
+| ``x`` | permit whitespace and line comments starting with ``#``             |
++-------+---------------------------------------------------------------------+
+| ``d`` | only ``\n`` is considered a line-terminator when using ``^``, ``$`` |
+|       | and ``.``                                                           |
++-------+---------------------------------------------------------------------+
+| ``g`` | keep matching until the end of ``source``, instead of stopping at   |
+|       | the first match.                                                    |
++-------+---------------------------------------------------------------------+
+
+
+Examples
+........
+
+In this example the ``pattern``does not match anything in the ``source`` and the result
+is an empty table:
+
+::
+
+    cr> select regexp_matches('foobar', '^(a(.+)z)$') as matched;
+    +---------+
+    | matched |
+    +---------+
+    +---------+
+    SELECT 0 rows in set (... sec)
+
+In this example we find the term that follows two digits:
```

Comment:
isn't this example about using the flags?

Reply:
a few examples to show users valid uses of the function. It happens to follow the flags section, but it is not part of the flags section